patches/systemd/0002-Add-support-for-systemd-boot-for-Managarm.patch

```diff
@@ -0,0 +1,651 @@
+@@ -5005,6 +5008,7 @@ int exec_invoke(
+                 }
+ #endif
+ 
++#ifndef __managarm__
+                 if (!cap_test_all(bset)) {
+                         r = capability_bounding_set_drop(bset, /* right_now= */ false);
+                         if (r < 0) {
+@@ -5012,6 +5016,7 @@ int exec_invoke(
+                                 return log_exec_error_errno(context, params, r, "Failed to drop capabilities: %m");
+                         }
+                 }
++#endif
+ 
+                 /* Ambient capabilities are cleared during setresuid() (in enforce_user()) even with
+                  * keep-caps set.
+@@ -5129,6 +5134,7 @@ int exec_invoke(
+                 }
+ #endif
+ 
++#ifndef __managarm__
+                 /* PR_GET_SECUREBITS is not privileged, while PR_SET_SECUREBITS is. So to suppress potential
+                  * EPERMs we'll try not to call PR_SET_SECUREBITS unless necessary. Setting securebits
+                  * requires CAP_SETPCAP. */
+@@ -5160,6 +5166,7 @@ int exec_invoke(
+                                 *exit_status = EXIT_NO_NEW_PRIVILEGES;
+                                 return log_exec_error_errno(context, params, errno, "Failed to disable new privileges: %m");
+                         }
++#endif
+ 
+ #if HAVE_SECCOMP
+                 r = apply_address_families(context, params);
+diff --git a/src/core/main.c b/src/core/main.c
+index 5b11c79..d79d48b 100644
+--- a/src/core/main.c
++++ b/src/core/main.c
+@@ -2380,7 +2380,10 @@ static int initialize_runtime(
+                         (void) hostname_setup(true);
+                         /* Force transient machine-id on first boot. */
+                         machine_id_setup(/* root= */ NULL, /* force_transient= */ first_boot, arg_machine_id, /* ret_machine_id */ NULL);
++// Managarm does not support loopback devices yet
++#ifndef __managarm__
+                         (void) loopback_setup();
++#endif
+                         bump_unix_max_dgram_qlen();
+                         bump_file_max_and_nr_open();
+                         test_usr();
+diff --git a/src/core/manager.c b/src/core/manager.c
+index 22b1822..3b74cb0 100644
+--- a/src/core/manager.c
++++ b/src/core/manager.c
+@@ -926,6 +926,7 @@ int manager_new(RuntimeScope runtime_scope, ManagerTestRunFlags test_run_flags,
+                 .executor_fd = -EBADF,
+         };
+ 
++
+         unit_defaults_init(&m->defaults, runtime_scope);
+ 
+ #if ENABLE_EFI
+diff --git a/src/core/mount.c b/src/core/mount.c
+index 66917df..82fa9dd 100644
+--- a/src/core/mount.c
++++ b/src/core/mount.c
+@@ -1553,6 +1553,11 @@ static void mount_sigchld_event(Unit *u, pid_t pid, int code, int status) {
+                         f == MOUNT_SUCCESS,
+                         code, status);
+ 
++#ifdef __managarm__
++        if(m->state == MOUNT_MOUNTING && f == MOUNT_SUCCESS)
++                m->state = MOUNT_MOUNTING_DONE;
++#endif
```

Comment:
Why is this done? We should add a comment.

Reply:
I've added a comment.